### ISSUE TITLE
sqlite3.IntegrityError: UNIQUE constraint failed: episode.podcast_id, episode.download_filename

### ISSUE THREAD

**Issue by galantra** [sairuk/gpodder_pc_import#2]
Hi there. Last time I tried to move to gPodder in February, but I couldn't get it wroking and so I gave up. Now I'm trying again, because I really want to move my podcasts to FOSS.

When I execute the new script, I get a couple of lines with `FOUND: `. Then it stops with this:

```
FOUND: default_tc.mp3, updating gpodder database
FOUND: default_tc.mp3, updating gpodder database
Traceback (most recent call last):
  File "main.py", line 93, in <module>
    main()
  File "main.py", line 76, in main
    conn.execute("UPDATE episode SET state = %d, is_new = %d, download_filename = '%s' WHERE id = '%d'" % (state, is_new, fname, id))
sqlite3.IntegrityError: UNIQUE constraint failed: episode.podcast_id, episode.download_filename
```

"IntegrityError" sound severe… Is this fixable? Maybe the script should just skip the folders where it finds an error. At least most podcasts would get updated.

**Comment by sairuk** [comment]
For whatever reason it found a duplicate filename and when it attempted to push the second name to the db it hit the unique constraint. Perhaps I need to revisit gpodder, i assume it handles duplicate names between podcasts in some fasion. I haven't looked at it since your last issue.

**Comment by sairuk** [comment]
I wouldn't expect this to be fixed anytime soon. packaged and/or latest gPodder doesn't actually run properly here so i'll need to look into fixing gPodder before I can fix this if i am so inclined

For the meantime you can identify the duplication and work around it by removing it from your tree, if thats not possible let me know.

**Comment by sairuk** [comment]
While i'm not entirely sure on the condition you hit based on your dir structure i wasn't specifically targeting the podcast_id during the import. I've made these changes now so this may fix your issue see 48e32584128f7345e210f2d4c5bfc9da0b96504f and let me know
